commands/get/environments.go

```diff
@@ -0,0 +1,120 @@
+package get
+
+//handle the environments list for rhmap.
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"text/template"
+
+	"github.com/feedhenry/rhm/commands"
+	"github.com/feedhenry/rhm/storage"
+	"github.com/urfave/cli"
+)
+
+//environmentsCmd constructs the required writer in order to send the response to the right place.
+type environmentsCmd struct {
+	out    io.Writer
+	in     io.Reader
+	getter func(*http.Request) (*http.Response, error)
+	store  storage.Storer
+}
+
+var environmentListTemplate = `
+{{range . }} 
+Id      |  {{ .ID}} 
+Label   |  {{ .Label}} 
+Enabled |  {{ .Enabled}}
+Target  |
+
+  -- Id     | {{ .Target.ID}}
+  -- Label  | {{ .Target.Label}}
+  -- Env    | {{ .Target.Env}}
+
+
+{{end}}        
+`
+
+//Environment Defines our cli command including its flags and usage then returns the command to allow a user to do specific operations on environments
+func (ec *environmentsCmd) Environments() cli.Command {
+	return cli.Command{
+		Name:        "environments",
+		Action:      ec.environmentsAction,
+		Usage:       "environments",
+		Description: "environments:where allows listing environments in rhm",
+	}
+}
+
+// environmentAction is where the logic is pulled together to perform the command. This funtion conforms to the cli action
+func (ec *environmentsCmd) environmentsAction(ctx *cli.Context) error {
+	var (
+		url = "%s/api/v2/environments/%s"
+	)
+	userData, err := ec.store.ReadUserData()
+	if err != nil {
+		return cli.NewExitError("could not read userData "+err.Error(), 1)
+	}
+	fullURL := fmt.Sprintf(url, userData.Host, "all")
+
+	newrequest, err := http.NewRequest("GET", fullURL, nil)
+	if err != nil {
+		return cli.NewExitError("could not create new request object "+err.Error(), 1)
+	}
+
+	// Use the fh-mbass-api-key
+	// I have hardcoded this for now - not ideal
+	// Looking into doing a query to get this key
+
+	newrequest.Header.Add("X-FH-AUTH-USER", "9052dfadcf8994d997e6d1fa7c395d760e448312")
```

Comment:
yeah maybe we could take an apiKey as part of the login command. Also kinda suprised this doesn't work with the cookie. What was the error you were getting back?